miden-lib/src/faucets/mod.rs

```diff
@@ -0,0 +1,113 @@
+use crate::{assembler::assembler, auth::AuthScheme};
+use miden_objects::{
+    accounts::{Account, AccountCode, AccountId, AccountStorage, AccountType, AccountVault},
+    assembly::ModuleAst,
+    crypto::merkle::MerkleStore,
+    utils::{
+        string::{String, ToString},
+        vec,
+    },
+    AccountError, Felt, StarkField, Word, ZERO,
+};
+
+/// Creates a new faucet account with basic faucet interface, specified authentication scheme,
+/// and provided meta data (token symbol, decimals, max supply).
+///
+/// The basic faucet interface exposes two procedures:
+/// - `distribute`, which mints an assets and create a note for the provided recipient.
+/// - `burn`, which burns the provided asset.
+///
+/// `distribute` requires authentication. The authentication procedure is defined by the specified
+/// authentication scheme.
+/// `burn` does not require authentication and can be called by anyone.
+/// Public key information for the scheme is stored in the account storage at slot 0.
+/// The token metadata is stored in the account storage at slot 1.
```

Comment:
minor nit: I think the formatting here will look a bit off when processed by the doc builder. I'd probably do something like this:
```Rust
/// `distribute` requires authentication. The authentication procedure is defined by the specified
/// authentication scheme. `burn` does not require authentication and can be called by anyone.
///
/// Public key information for the scheme is stored in the account storage at slot 0. The token
/// metadata is stored in the account storage at slot 1.
```